config_test.go

```diff
@@ -0,0 +1,636 @@
+				"value":  "r00t",
+				"public": false,
+			},
+		}
+		payload, _ := json.Marshal(rawPayload)
+		server.prepareResponse(preparedResponse{
+			method:  "GET",
+			path:    "/apps/" + appName + "/env",
+			code:    http.StatusOK,
+			payload: payload,
+		})
+	}
+	cleanup, err := setupFakeTarget(server.url())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+	var c projectConfigGet
+	err = c.Flags().Parse(true, []string{
+		"-n", "proj1",
+		"-e", "dev,stage,prod",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	var stdout, stderr bytes.Buffer
+	ctx := cmd.Context{Stdout: &stdout, Stderr: &stderr}
+	client := cmd.NewClient(http.DefaultClient, &ctx, &cmd.Manager{})
+	err = c.Run(&ctx, client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedOutput := `config vars in "dev":
+
+ APP_NAME=proj1-dev
+ USER_NAME=root
+ USER_PASSWORD=*** (private config)
+
+
+config vars in "prod":
+
+ APP_NAME=proj1-prod
+ USER_NAME=root
+ USER_PASSWORD=*** (private config)
+
+
+`
+	if stdout.String() != expectedOutput {
+		t.Errorf("wrong output\nwant:\n%q\ngot:\n%q", expectedOutput, stdout.String())
+	}
+	expectedStderr := `WARNING: project not found in environment "stage"` + "\n"
+	if stderr.String() != expectedStderr {
+		t.Errorf("wrong error output\nwant:\n%q\ngot:\n%q", expectedStderr, stderr.String())
+	}
+}
+
+func TestProjectConfigGetErrorInOneOfTheApps(t *testing.T) {
+	server := newFakeServer(t)
+	defer server.stop()
+	server.prepareResponse(preparedResponse{
+		method:  "GET",
+		path:    "/apps/proj1-stage/env",
+		code:    http.StatusInternalServerError,
+		payload: []byte("something went wrong"),
+	})
+	appNames := []string{"proj1-dev", "proj1-prod"}
+	for _, appName := range appNames {
+		rawPayload := []map[string]interface{}{
+			{
+				"name":   "APP_NAME",
+				"value":  appName,
+				"public": true,
+			},
+			{
+				"name":   "USER_NAME",
+				"value":  "root",
+				"public": true,
+			},
+			{
+				"name":   "USER_PASSWORD",
+				"value":  "r00t",
+				"public": false,
+			},
+		}
+		payload, _ := json.Marshal(rawPayload)
+		server.prepareResponse(preparedResponse{
+			method:  "GET",
+			path:    "/apps/" + appName + "/env",
+			code:    http.StatusOK,
+			payload: payload,
+		})
+	}
+	cleanup, err := setupFakeTarget(server.url())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+	var c projectConfigGet
+	err = c.Flags().Parse(true, []string{
+		"-n", "proj1",
+		"-e", "dev,stage,prod",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	var stdout, stderr bytes.Buffer
+	ctx := cmd.Context{Stdout: &stdout, Stderr: &stderr}
+	client := cmd.NewClient(http.DefaultClient, &ctx, &cmd.Manager{})
+	err = c.Run(&ctx, client)
+	if err == nil {
+		t.Fatal("unexpected <nil> error")
+	}
+	expectedOutput := `config vars in "dev":
+
+ APP_NAME=proj1-dev
+ USER_NAME=root
+ USER_PASSWORD=*** (private config)
+
+
+`
+	if stdout.String() != expectedOutput {
+		t.Errorf("wrong output\nwant:\n%q\ngot:\n%q", expectedOutput, stdout.String())
+	}
```

Comment:
The command framework will wrap the error returned by the function and write it to stderr. I'll update the code to continue and only return the error at the end.